test/forward_backward_compatibility/check_forward_backward_compatibility.py

```diff
@@ -302,6 +302,12 @@
     ("aten::reduce_scatter_tensor", datetime.date(9999, 1, 30)),
     ("aten::all_gather_into_tensor", datetime.date(9999, 1, 30)),
     ("aten::all_reduce", datetime.date(9999, 1, 30)),
+    ("aten::to_sparse.out", datetime.date(2023, 6, 30)),
+    ("aten::to_sparse.sparse_dim_out", datetime.date(2023, 6, 30)),
+    ("aten::to_sparse_bsc.out", datetime.date(2023, 6, 30)),
+    ("aten::to_sparse_bsr.out", datetime.date(2023, 6, 30)),
+    ("aten::to_sparse_csc.out", datetime.date(2023, 6, 30)),
+    ("aten::to_sparse_csr.out", datetime.date(2023, 6, 30)),
```

Comment:
Should these dates be updated, say, increment by half a year?

Reply:
Changed.

Reply:
One more comment here:  I've modeled this PR according to the `to_dense()`/`_to_dense()` separation.  Here, in `native_functions.yaml` file  `to_dense` is written so that `to_dense.out` is not generated, and this is the reason I've removed `.out` versions for `to_sparse*()` methods.  But it could be a backward incompatible change in case someone used this.  Please advise if it would be preferable to keep these.